libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -105,23 +106,25 @@ export class FlagdWebProvider implements Provider {
     };
   }
 
-  private async retryConnect(context: EvaluationContext) {
+  private async retryConnect() {
     this._delayMs = Math.min(this._delayMs * BACK_OFF_MULTIPLIER, this._maxDelay);
 
     return new Promise<void>((resolve) => {
       this._callbackClient.eventStream(
         {},
         (message) => {
+          // get the context at the time of the message
+          const currentContext = OpenFeature.getContext();
           this._logger?.debug(`${FlagdWebProvider.name}: event received: ${message.type}`);
           switch (message.type) {
             case EVENT_PROVIDER_READY:
-              this.fetchAll(context).then(() => {
```

Comment:
This context was basically cached in this closure, which would cause `resolveAll` to resolve with an out-of-date context in some cases.